src/parameterized_tendencies/gravity_wave_drag/non_orographic_gravity_wave.jl

```diff
@@ -242,36 +302,40 @@ function non_orographic_gravity_wave_tendency!(
 end
 
 function non_orographic_gravity_wave_forcing(
-    ᶜu,
-    ᶜbf,
-    ᶜρ,
-    ᶜz,
+    old_ᶜu,
+    old_ᶜbf,
+    old_ᶜρ,
+    old_ᶜz,
     source_level,
     damp_level,
     source_ampl,
     Bw,
     Bn,
+    B0,
     cw,
     cn,
     flag,
     c,
     c0,
     nk,
 )
-    FT = eltype(ᶜz)
+    FT = eltype(old_ᶜz)
     # add an extra layer above model top so that forcing between the very top
     # model layer and the upper boundary can be calculated
-    append!(ᶜu, FT(2) * ᶜu[end] - ᶜu[end - 1])
-    append!(ᶜρ, ᶜρ[end] * ᶜρ[end] / ᶜρ[end - 1])
-    append!(ᶜbf, ᶜbf[end])
-    append!(ᶜz, FT(2) * ᶜz[end] - ᶜz[end - 1])
+    ᶜu = vcat(old_ᶜu, FT(2) * old_ᶜu[end] - old_ᶜu[end - 1])
+    ᶜρ = vcat(old_ᶜρ, old_ᶜρ[end] * old_ᶜρ[end] / old_ᶜρ[end - 1])
+    ᶜbf = vcat(old_ᶜbf, old_ᶜbf[end])
+    ᶜz = vcat(old_ᶜz, FT(2) * old_ᶜz[end] - old_ᶜz[end - 1])
 
+    
     # wave spectra and the source amplitude
     nc = length(c)
     c_hat0 = c .- ᶜu[source_level] # c0mu0
-    Bw_exp = @. exp(-log(2.0) * ((c * flag + c_hat0 * (1 - flag) - c0) / cw)^2)
-    Bn_exp = @. exp(-log(2.0) * ((c * flag + c_hat0 * (1 - flag) - c0) / cn)^2)
-    B0 = @. sign(c_hat0) * (Bw * Bw_exp + Bn * Bn_exp)
+   
+    B0 = @. sign(c_hat0) * (
+        Bw * exp(-log(2.0) * ((c * flag + c_hat0 * (1 - flag) - c0) / cw)^2) + 
+        Bn * exp(-log(2.0) * ((c * flag + c_hat0 * (1 - flag) - c0) / cn)^2)
+        )
```

Comment:
```suggestion
    @. B0 = sign(c_hat0) * (
        Bw * exp(-log(2.0) * ((c * flag + c_hat0 * (1 - flag) - c0) / cw)^2) + 
        Bn * exp(-log(2.0) * ((c * flag + c_hat0 * (1 - flag) - c0) / cn)^2)
        )
```
The form `B0 = @. sign(c_hat0) ...` points `B0` to new memory. the form I suggested will mutate `B0` in-place